fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedSegmentInfoFormat.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * LuceneOptimizedSegmentInfoFormat.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import org.apache.lucene.codecs.SegmentInfoFormat;
+import org.apache.lucene.codecs.lucene70.Lucene70SegmentInfoFormat;
+import org.apache.lucene.index.IndexFileNames;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class LuceneOptimizedSegmentInfoFormat extends SegmentInfoFormat {
+    private final SegmentInfoFormat segmentInfosFormat = new Lucene70SegmentInfoFormat();
+
+    public LuceneOptimizedSegmentInfoFormat() {
+    }
+
+    @Override
+    public SegmentInfo read(final Directory directory, final String segmentName, final byte[] segmentID, final IOContext context) throws IOException {
+        SegmentInfo segmentInfo = segmentInfosFormat.read(new LuceneOptimizedWrappedDirectory(directory), segmentName, segmentID, context);
+        SegmentInfo segmentInfoToReturn = new SegmentInfo(directory, segmentInfo.getVersion(), segmentInfo.getMinVersion(), segmentInfo.name,
```

Comment:
Yes but the SegmentInfoFormat from lucene does things with that directory that we want to bypass. We use the new directory then write the files we want to include.

Reply:
@sophiabuell yes that sounds right. My understanding is that is why we pass a ```new LuceneOptimizedWrappedDirectory(directory)``` here instead of the one from Lucene. But I am wondering why the ```segmentInfoToReturn``` still wants to use the one passed in from Lucene?

Reply:
Comment it out?  You will see that Lucene has a check to make sure the current directory (FDBDirectory) matches the one on SegmentInfo.